### ISSUE TITLE
Issue linting with macro at top of file in 2.0.0a6

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#4485]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Attempting to lint a file with a macro at the top in dbt project

### Expected Behaviour

No issues/failures

### Observed Behaviour

Lint the file produced this error message: `NotImplementedError: Unexpected templated indent. Report this as a bug on GitHub. Segment: <WhitespaceSegment: ([L:  1, P:  1]) '    '>`

### How to reproduce

Within a dbt project, create the following file

`models/test.sql`
```sql
{{ dev_config() }}

select col_1 from my_table
```

Where `dev_config` is something like this:

`macros/dev_config.sql`
```sql
{% macro dev_config() %}
    {% if var('dev', false) %}
        {{ config(materialized='table') }}
    {% endif %}
{% endmacro %}
```

Then lint
```sh
$ sqlfluff lint model/test.sql
```

### Dialect

ANSI

### Version

2.0.0a6

### Configuration

`.sqlfluff`
```
[sqlfluff]
dialect = ansi
templater = dbt
# aliasing.table: oracle doesn't do explicit table aliases
# structure.column_order: we want columns in a particular order, to make output easier to validate
exclude_rules = aliasing.table,structure.column_order
max_line_length = 125

[sqlfluff:templater:dbt]
profiles_dir = .

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:references.keywords]
ignore_words = procedures
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Yep - took a while, but I've managed to recreate this on my side. Found the issue - simple fix. Making a simple test case is less hard. PR will be up soon.
